### PR TITLE
Command-Line Argument Parsing for EXAONE

### PR DESCRIPTION
**Pull request by jh941213**
## Purpose

* Implements command-line argument parsing for LG AI EXAONE connector
* Adds `LGArgumentOptions` class that inherits from `ArgumentOptions` base class
* Enables parsing of `--base-url` and `--model` command-line arguments for LG AI EXAONE
* Integrates LG argument parsing into the main `ArgumentOptions.Parse` method
* Supports configuration priority: CLI arguments > Configuration file settings

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] New feature
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## README updated?

The top-level readme for this repo contains a link to each sample in the repo. If you're adding a new sample did you update the readme?

```
[ ] Yes
[x] No
[ ] N/A
```

## How to Test
*  Get the code

```
git clone https://github.com/jh941213/open-chat-playground.git
cd open-chat-playground
git checkout feature/251
```

* Test the code

```
dotnet test --filter Category=UnitTest
```

## What to Check
Verify that the following are valid
* Check if the command line arguments include `--base-url` and `--model` for LG AI EXAONE
* Verify that `LGArgumentOptions` class inherits from `ArgumentOptions`
* Confirm that LG arguments are properly parsed and integrated into `AppSettings`
* Ensure all existing tests continue to pass (289 tests should succeed)
* Validate that LG-specific tests cover configuration priority and edge cases
* Check that LG connector is positioned correctly in the Vendor section (after Anthropic, before Naver/OpenAI)

### CODE REVIEW

File: src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs [hunk 1]
```diff
@@ -144,6 +146,7 @@ public static AppSettings Parse(IConfiguration config, string[] args)
 
         switch (options)
         {
+            // MaaS providers
```
Comment (sikutisa) [inline src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs:149]: 이 주석을 추가하신 이유가 있나요?

File: src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs [hunk 2]
```diff
@@ -158,6 +161,13 @@ public static AppSettings Parse(IConfiguration config, string[] args)
                 settings.GitHubModels.Model = github.Model ?? settings.GitHubModels.Model;
                 break;
 
+            case GoogleVertexAIArgumentOptions googleVertexAI:
+                settings.GoogleVertexAI ??= new GoogleVertexAISettings();
+                settings.GoogleVertexAI.ApiKey = googleVertexAI.ApiKey ?? settings.GoogleVertexAI.ApiKey;
+                settings.GoogleVertexAI.Model = googleVertexAI.Model ?? settings.GoogleVertexAI.Model;
+                break;
+
+            // Local providers
```
Comment (sikutisa) [inline src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs:170]: task 범위 밖의 수정사항입니다.

File: src/OpenChat.PlaygroundApp/Options/LGArgumentOptions.cs
Comment (sikutisa) [inline src/OpenChat.PlaygroundApp/Options/LGArgumentOptions.cs:1]: 이 파일이 `ArgumentOptions`를 상속받는지 확인하는 테스트를 추가해주세요. 아래 메서드를 확인해보세요. 
https://github.com/aliencube/open-chat-playground/blob/360f30cd709294863b7aa920a2e685d15e1e1ac8/test/OpenChat.PlaygroundApp.Tests/Abstractions/ArgumentOptionsTests.cs#L297

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        settings.LG.ShouldNotBeNull();
+        settings.LG.BaseUrl.ShouldBe(cliBaseUrl);
+        settings.LG.Model.ShouldBe(cliModel);
+    }
+
+
+    [Trait("Category", "UnitTest")]
+    [Theory]
+    [InlineData("https://cli.lg-exaone/api", "cli-model")]
+    public void Given_LG_With_KnownArguments_When_Parse_Invoked_Then_Help_ShouldBe_False(string cliBaseUrl, string cliModel)
```
Comment (sikutisa) [inline test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:203]: `ShouldBe` -> `Should_Be`

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+
+        // Assert
+        settings.Help.ShouldBeFalse();
+    }
+
+    [Trait("Category", "UnitTest")]
+    [Theory]
+    [InlineData("--base-url")]
+    [InlineData("--model")]
+    public void Given_LG_With_KnownArgument_WithoutValue_When_Parse_Invoked_Then_Help_ShouldBe_False(string argument)
```
Comment (sikutisa) [inline test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:220]: `ShouldBe` -> `Should_Be`

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        var settings = ArgumentOptions.Parse(config, args);
+
+        // Assert
+        settings.Help.ShouldBeFalse();
+    }
+
+    [Trait("Category", "UnitTest")]
+    [Theory]
+    [InlineData("https://cli.lg-exaone/api", "--unknown-flag")]
+    public void Given_LG_With_Known_And_Unknown_Argument_When_Parse_Invoked_Then_Help_ShouldBe_True(string cliBaseUrl, string argument)
```
Comment (sikutisa) [inline test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:236]: `ShouldBe` -> `Should_Be`

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+        // Arrange
+        var config = BuildConfigWithLG(BaseUrl, Model);
+        var args = new[] { "--base-url", cliBaseUrl, "--model", cliModel };
+
+        // Act
+        var settings = ArgumentOptions.Parse(config, args);
+
+        // Assert
+        settings.Help.ShouldBeFalse();
+    }
```
Comment (sikutisa) [inline test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:214]: 이 테스트의 의미를 잘 모르겠습니다. 
250라인부터 있는 `Given_CLI_Only_When_Parse_Invoked_Then_Help_Should_Be_False`과 구분해서 작성한 이유가 있나요?

**Review by sikutisa — CHANGES_REQUESTED** [review]

File: src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs (hunk 2 above)
Comment (tae0y) [reply src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs:170]: #439 가 머지되며 conflict가 생겼습니다.
main을 머지하여 해소해주셔요.

File: src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs (hunk 1 above)
Comment (jh941213) [reply src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs:149]: 불필요한 주석이라 제거하겠습니다..!

File: src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs (hunk 2 above)
Comment (jh941213) [reply src/OpenChat.PlaygroundApp/Abstractions/ArgumentOptions.cs:170]: @sikutisa 맞습니다. 권한 밖의 수정을 했네요 수정하고 , rebase conflict 하겠습니다!!

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs (hunk 5 above)
Comment (jh941213) [reply test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:236]: 수정하겠습니다..!

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs (hunk 6 above)
Comment (jh941213) [reply test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:214]: 중복되고있는 테스트라 삭제 하겠습니다 오버라이드  테스트인데 크게 불필요해보이네요

File: src/OpenChat.PlaygroundApp/Options/LGArgumentOptions.cs
Comment (sikutisa) [reply src/OpenChat.PlaygroundApp/Options/LGArgumentOptions.cs:1]: > 이 파일이 `ArgumentOptions`를 상속받는지 확인하는 테스트를 추가해주세요. 아래 메서드를 확인해보세요.
> 
> https://github.com/aliencube/open-chat-playground/blob/360f30cd709294863b7aa920a2e685d15e1e1ac8/test/OpenChat.PlaygroundApp.Tests/Abstractions/ArgumentOptionsTests.cs#L297

이 부분이 수정이 안 됐습니다.

File: test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+        var args = new[] { "--base-url", cliBaseUrl, argument };
+
+        // Act
+        var settings = ArgumentOptions.Parse(config, args);
+
+        // Assert
+        settings.Help.ShouldBeTrue();
+    }
+
+
```
[the remark this answers is not in the document]
Comment (jh941213) [reply test/OpenChat.PlaygroundApp.Tests/Options/LGArgumentOptionsTests.cs:249]: 네 수정 하겠습니다!
